common/src/reserved.rs

```diff
@@ -36,18 +36,21 @@ impl ReservedState {
                     .or_insert(member.consensus_voting_power);
             }
         }
-        // TODO handle error
-        Ok(self
-            .consensus_leader_order
-            .iter()
-            .map(|name| (self.query_public_key(name).unwrap(), validator_set[name]))
-            .collect())
+        let mut result = Vec::new();
+        for (name, voting_power) in validator_set {
+            let public_key = self.query_public_key(&name).ok_or_else(|| {
+                format!("The public key of {name} is not found in the genesis info.")
```

Comment:
This comment contains wrong information. The public key is searched from the reserved state.
```suggestion
                format!("The public key of {name} is not found in the reserved state.")
```

Reply:
would change it.